### ISSUE TITLE
Training

### ISSUE THREAD

**Issue by Zumbalamambo** [texastony/adapting-c3d-keras#1]
How do I train it in my own dataset?

**Comment by texastony** [comment]
That would depend on how many GPUs you have access to and the dimensions of your data. But if your video is already 112 x 112, and you only have one GPU (or none), you can use the Catar class as a training iterator that will group frames into chunks of 16, associate them with the correct label, and then feed it to the the network. You will need to download the weights via running: adapting-c3d-keras/models/get_weights_and_mean.sh. Then you can use Keras' load model from json, followed by `.load_weights`. 

Then, I would delete the top layer, and set the 2nd to last layer (model.layers[-2]) to be the output layer. Run all of your data through this model once, and save the output to disk (should take forever with out multiple GPUs or an HPC). Train a softmax layer on these saved outputs for a while. Then you can append this trained softmax layer back on the original model and fine tune the whole thing if you desire. 

I should note, I worked with C3D for a while, and made some progress towards labeling activity in novel case. However, I ultimately found inception-net run on single frames combined with a kalman filter worked much better (92% accuracy in my fairly trivial case). I am working on a post about this, but its being delayed by my full plate... Let me know if you have any other questions.
